### PR TITLE
build: support multi-entry build

### PR DESCRIPTION
**Pull request by micmro**
Hi @brendanrygus 
I got the multi-entry build working, can you check if this works for your needs?

Screen Shot 2020-10-24 at 9 55 14 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +62,8 @@
   },
   "scripts": {
     "start": "tsdx watch",
-    "build": "tsdx build",
+    "build": "tsdx build --entry src/index.tsx --entry src/register.tsx && npm run post-build",
+    "post-build": "node scripts/post-tsdx.js",
```
Comment (micmro) [inline package.json:66]: needed a post tsdx step to fix the cjs enty files (`dist/index.js` points at the wrong files and `dist/register.js` does not exist)

File: package.json [hunk 2]
```diff
@@ -38,7 +38,17 @@
...
+  "exports": {
+    ".": {
+      "import": "./dist/index.esm.js",
+      "require": "./dist/index.js"
+    },
+    "./register": {
+      "import": "./dist/register.esm.js",
+      "require": "./dist/register.js"
+    }
+  },
```
Comment (micmro) [inline package.json:51]: I have not tested this out yet, https://nodejs.org/api/packages.html#packages_writing_dual_packages_while_avoiding_or_minimizing_hazards might be useful but `"type": "module"` does not play well with TSDX

File: tsdx.config.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// TSDX does not currently support multiple entry files out of the box,
```
Comment (micmro) [inline tsdx.config.js:1]: TSDX does not currently support multiple entry files out of the box, so I had to overwrite the Rollup config
Ref: https://github.com/formium/tsdx/issues/175#issuecomment-564814325

**Comment by lifeiscontent** [comment]
@micmro I was thinking of just tossing out tsdx all together and just creating a tsc build with esm and cjs outputs. wdyt?

**Comment by micmro** [comment]
Makes sense, tsdx works great for simple projects but is's quite limiting. I'll just need find time for this.

**Comment by brendanrygus** [comment]
In the event that we end up keeping this approach...

I finally tested this out tonight (sorry for the delay!) but had some issues implementing it in the test app. I could only get Webpack to resolve the import from `"storybook-addon-apollo-client/dist/register"`, and not from `"storybook-addon-apollo-client/register"`. 

I'm currently installing it from a local directory `"storybook-addon-apollo-client": "file:../storybook-addon-apollo-client"` if that makes an impact on the resolution? I'll hopefully get time to look into this some more soon, and keep troubleshooting

**Comment by lifeiscontent** [comment]
@brendanrygus I'm pretty sure storybook resolves modern ES6 code, so we could potentially just use tsc and have it build for ES6

**Pull request closed without merging**